src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -438,73 +438,43 @@ internal void UpdateWithExemplar(long number, ReadOnlySpan<KeyValuePair<string,
         {
             case AggregationType.LongSumIncomingDelta:
                 {
-                    var sw = default(SpinWait);
-                    while (true)
+                    AcquireLock(ref this.mpComponents!.IsCriticalSectionOccupied);
+
+                    unchecked
                     {
-                        if (Interlocked.Exchange(ref this.mpComponents!.IsCriticalSectionOccupied, 1) == 0)
-                        {
-                            // Lock acquired
-                            unchecked
-                            {
-                                this.runningValue.AsLong += number;
-                            }
-
-                            if (isSampled)
-                            {
-                                this.mpComponents.ExemplarReservoir.Offer(number, tags);
-                            }
-
-                            // Release lock
-                            Interlocked.Exchange(ref this.mpComponents.IsCriticalSectionOccupied, 0);
-                            break;
-                        }
+                        this.runningValue.AsLong += number;
+                    }
 
-                        sw.SpinOnce();
+                    if (isSampled)
+                    {
+                        this.mpComponents.ExemplarReservoir.Offer(number, tags);
```

Comment:
Is it possible this exemplar code will throw? Not something we have to deal with on this PR, but we would be in a bad state if we exited without releasing the lock.

Reply:
Good catch. We have marked `ExemplarReservoir` internal for now and the two implementations that are present in this repo would not throw an exception. This might be an issue when we mark `ExemplarReservoir` public and users bring in their custom implementations of `ExemplarReservoir.Offer`.

I have added TODO comments for these for now.